web/samples/attachment-api/utilities.js

```diff
@@ -0,0 +1,206 @@
+		disableBtn.value   = 'Disable';
+		div.appendChild(disableBtn);
+		
+		var enabledLabel = document.createElement("a");
+		enabledLabel.id	= 'enablement_' + elemId;
+		enabledLabel.textContent = " Currently enabled.";
+		
+		// We must rely on MutationObservers to stay current about enablement.
+		var observationTarget = elem;
+		var observationConfig = {attributes: true, attributeFilter: ['class']};
+		
+		var enablementObserver = new MutationObserver(function(mutations) {
+			var isDisabled = mutations[0].target.className.indexOf('kmw-disabled') >= 0;
+			enabledLabel.textContent = (isDisabled ? " Currently disabled." : " Currently enabled.");
+		});
+		enablementObserver.observe(observationTarget, observationConfig);
+		
+		div.appendChild(enabledLabel);
+		div.appendChild(document.createElement("p"));
+		
+		if(elem.tagName.toLowerCase() != "iframe") { 
+			// Set Keyboard button
+			var setBtn = document.createElement("input");
+			setBtn.type 		= 'button';
+			setBtn.id   		= 'set_' + elemId;
+			setBtn.onclick = function() {
+				kmw.setKeyboardForControl(document.getElementById(elemId), 'dzongkha', 'dzo');
+				
+				if(kmw.isAttached(elem)) {
+					kbdLabel.textContent = " Using independently-tracked keyboard.";
+				}
+			}; 
+			setBtn.value   = 'Set to Dzongkha';
+			div.appendChild(setBtn);
+			
+			// Clear Keyboard button
+			var clearBtn = document.createElement("input");
+			clearBtn.type 		= 'button';
+			clearBtn.id   		= 'clear_' + elemId;
+			clearBtn.onclick = function() {
+				kmw.setKeyboardForControl(document.getElementById(elemId), null, null);
+				
+				if(kmw.isAttached(elem)) {
+					kbdLabel.textContent = " Using globally-selected keyboard.";
+				}
+			}; 
+			clearBtn.value   = 'Clear Keyboard';
+			div.appendChild(clearBtn);
+			
+			var kbdLabel = document.createElement("a");
+			kbdLabel.id	= 'independence_' + elemId;
+			window.setTimeout(function () {
+				if(kmw.isAttached(elem)) {
+					var attached = kmw.isAttached(elem);
+					kbdLabel.textContent = attached ? " Using globally-selected keyboard." : '';
+				}
+			}, 1);
+			div.appendChild(kbdLabel);
+		}
+		
+		return div;
+	}
+  
+	function addInput() {
+	  var masterDiv = document.getElementById('DynamicInputs');
+	  var newInput = document.createElement("input");
+		var i = inputCounter++;
+		
+	  newInput.id = 'input' + i;
+	  newInput.className = 'test';
+	  newInput.placeholder = "Dynamic area #" + i + "!";
+	  
+	  var newDiv = generateDiagnosticDiv(newInput);
+	  
+	  masterDiv.appendChild(newDiv);
+  }
+	
+  function addText() {
+	  var masterDiv = document.getElementById('DynamicTextareas');
+	  var newTextArea = document.createElement("textarea");
+	  var i = inputCounter++;
+	  
+	  newTextArea.id = 'textarea' + i;
+	  newTextArea.className = 'test';
+	  newTextArea.placeholder = "Dynamic area #" + i + "!";
+	  
+	  var newDiv = generateDiagnosticDiv(newTextArea);
+	  
+	  masterDiv.appendChild(newDiv);
+  }
+  
+  function addIFrame() {
```

Comment:
I did some independent tests at one point in development, although that doesn't seem to be present on the page as of yet.  I'll see if I can add it something for that to the testing page without much issue.